cli/src/semgrep/formatter/text.py

```diff
@@ -554,7 +554,8 @@ def print_text_output(
     last_file = None
     last_rule_id = None
     last_message = None
-    sorted_rule_matches = sorted(rule_matches, key=lambda r: (r.path, r.rule_id))
+    # Findings come in pre-sorted.
+    sorted_rule_matches: List[RuleMatch] = list(rule_matches)
```

Comment:
Actually, this was an attempt to sort the results by path first. We should do this (but correctly - I don't think it was correct). It's cumbersome because the findings are grouped by rule until the very end.

Reply:
done